.github/workflows/go-presubmit.yaml

```diff
@@ -64,3 +64,31 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
       - name: build
         run: make vendor && make build
+  
+  e2e:
+    name: e2e
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          path: go/src/open-cluster-management.io/multicluster-controlplane
+      - name: install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: build image
+        run: make image
+        env:
+          IMAGE_NAME: quay.io/open-cluster-management/multicluster-controlplane:latest
+      - name: setup controlplanes on KinD cluster
+        run: |
+          make setup-controlplane
+        env: 
+          CONTROLPLANE_IMAGE: quay.io/open-cluster-management/multicluster-controlplane:latest
```

Comment:
You need to load the built image into KinD. in the current way, it is always using quay.io image instead of the image you have built in previous step.

Reply:
Actually, We don't need to add the **load the built image into KinD** here. I did the following steps to verify that.

Step 1: Generate a [unique image tag](https://github.com/yanmxa/multicluster-controlplane/blob/9fcffd285519e01b0fe35eeb81d5aca21d80c2df/.github/workflows/go-presubmit.yaml#L82) in the steps of the e2e job
Step 2: Build the controlplane image  with the tag `985967e`
  ```yaml
   ...
  make image
  shell: /usr/bin/bash -e {0}
  env:
    GO_VERSION: 1.19
    GO_REQUIRED_MIN_VERSION: 
    GOPATH: /home/runner/work/multicluster-controlplane/go
    IMAGE_NAME: quay.io/open-cluster-management/multicluster-controlplane:985967e
    ...
    Successfully built fdc185a87699
    Successfully tagged quay.io/open-cluster-management/multicluster-controlplane:985967e 
  ```
Step 3: Deploy controlplane with the specific image tag `985967e`
  ```yaml
  make setup-controlplane
  shell: /usr/bin/bash -e {0}
  env:
    GO_VERSION: 1.19
    GO_REQUIRED_MIN_VERSION: 
    GOPATH: /home/runner/work/multicluster-controlplane/go
    CONTROLPLANE_IMAGE: quay.io/open-cluster-management/multicluster-controlplane:985967e
    ...
 2022/12/01 04:20:13 [INFO] encoded CSR
2022/12/01 04:20:13 [INFO] signed certificate with serial number [164](https://github.com/open-cluster-management-io/multicluster-controlplane/actions/runs/3589423759/jobs/6041839419#step:6:165)776366242742668238015889854750638757494767851
## Using the Controlplane Image: quay.io/open-cluster-management/multicluster-controlplane:985967e
secret/controlplane-cert created
secret/controlplane-kubeconfig created
service/multicluster-controlplane created 
    ...
  ```
  Step 4: Verify the deployment is using the `CONTROLPLANE_IMAGE` image(this step in verified in my VM)
  Step 5: Verify the e2e tests are all passes.